### ISSUE TITLE
Think about allowing for HTML UI early on for plugin scripts

### ISSUE THREAD

**Issue by bmix** [fman-community/fman#145]
Sublime now has HTML views. It may make sense to offer some techology to use HTML as a user-interface for creation of additional dialogs or the like. Users may want to create special property dialogs for different types of files, that are better than the OS default (especially valid on Windows). Or a plugin, that allows for mass-editing of MP3 attributes, etc. Makes me think of #76. AFAIK Qt has some powerful HTML view, no? Isn't Calibre using it in it's eBook editor? A default CSS would need to be created, that resembles fman's default UI style.

**Comment by mherrmann** [comment]
I think you're referring to Sublime's Phantoms? I don't know them well enough but don't think they're fully fledged HTML views. I think they're more like little snippets.

Qt has some support for HTML. fman inherits it. For example, fman's API function `show_status_message(...)` lets you include HTML like `<a href="http://...">link</a>.`

The problem with Qt and HTML is that Qt's support for style sheets is very lacking. Qt has a variant of CSS called "QSS". It has the same purpose and syntax. But it doesn't work like you'd expect from CSS at all, eg. when it comes to descendants of an element inheriting the parent's values. It also uses different properties.

In short, Qt has some support for HTML/CSS but it behaves very differently from current browsers. So it's unfortunately not like I could simply include some magic Qt view (except for maybe the extremely heavy `QWebView`, which is a browser in its own right).

What's more, I'm opposed to the idea of having too much "UI-stuff" in fman plugins. If it's so complex that it requires its own GUI, write it as a separate application and launch that from fman.

**Comment by mherrmann** [comment]
Regarding the default CSS you mention: Something like this will be implemented in #45. I'm closing the present issue for now.
